src/test/java/com/recurly/v3/QueryParamsTest.java

```diff
@@ -0,0 +1,39 @@
+package com.recurly.v3;
+
+import org.junit.jupiter.api.Test;
+
+import javax.management.Query;
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class QueryParamsTest {
+  @Test
+  public void testConstructor() {
+    QueryParams qp = new QueryParams();
+    assertEquals(qp.getParams(), new HashMap<String, Object>());
+  }
+
+  @Test
+  public void testSetParams() {
+    QueryParams qp = new QueryParams();
+    HashMap<String, Object> hm = new HashMap<>();
+    hm.put("test", 1);
+    qp.setParams(hm);
+    assertEquals(qp.getParams().get("test"), 1);
+  }
+
+  @Test
+  public void testAdd() {
+    QueryParams qp = new QueryParams();
+    qp.add("anothertest", 2);
```

Comment:
Do we want `add` to be public to the programmer?

Reply:
I thought we did in case we wanted to allow for other arbitrary query params that aren't specified by the API? If not, I can make it private and remove the test

Reply:
Oh, you know what? The programmer should be able to add arbitrary query params by specifying them in the HashMap and then setting them with `setParams`, so even if we made `add` private, it wouldn't block them from doing that. I still think it's a cool little utility method, but they could work around it by using `getParams`, adding to the HashMap and then setting it again with `setParams`.

I'll go ahead and make it private for now, but if someone wants us to make it public someday, I think we ought to go ahead and give them that.

Reply:
I'm worried people might use this over the named / typed methods. Also, if something is missing, that needs to be addressed in the generator.

Reply:
Suppose someone does `params.add("limit", 200)` for instance, and we change the name of "limit" to "pageSize" in the next version. They compiler can't help them.

Reply:
Yeah, that makes sense. Then should `setParams` still be public?

Reply:
Ideally, no.

Reply:
`add` should only be available internally

Reply:
Ok. The params hashmap is created when the object is instantiated. `getParams` will need to be public still, but I don't think anything actually uses `setParams` at this time. I might be able to just remove it altogether if we're not trying to provide that interface to the programmer